### PR TITLE
feat: add new ergo sdks to monorepo

### PR DESCRIPTION
**Pull request by dr-frmr**
## Summary by CodeRabbit

- **New Features**
  - Introduced official Python and TypeScript SDKs for the Honcho conversational memory platform, supporting peer, session, messaging, metadata, and context management.
  - Added synchronous and asynchronous client options in Python, and a high-level ergonomic client in TypeScript.
  - Provided paginated data handling, search functionality, and conversion of conversation context to OpenAI and Anthropic formats.
  - Included comprehensive example scripts for both SDKs demonstrating key workflows.
  - Added session peer configuration options to control observation settings.
  - Enabled multi-perspective chat queries and scoped session context retrieval with token limits and summaries.

- **Documentation**
  - Added detailed README files for both Python and TypeScript SDKs, including installation, usage examples, and links to further resources.
  - Provided extensive documentation for test suites and example scripts.
  - Added README for TypeScript SDK test suite explaining test coverage and patterns.

- **Tests**
  - Added thorough unit and integration test suites for the TypeScript SDK, covering all major classes, workflows, error handling, and edge cases.
  - Included test configuration and setup files.
  - Added comprehensive asynchronous and synchronous tests for Python SDK clients covering peers, sessions, metadata, messaging, search, and context.
  - Introduced mocks for core API clients to isolate tests.

- **Chores**
  - Added project configuration files such as `.gitignore`, `pyproject.toml`, `package.json`, `tsconfig.json`, and Jest configuration for streamlined development and publishing.
  - Updated internal code to ensure workspace existence before session creation and improved peer configuration management in sessions.

### CODE REVIEW

**Review by VVoruganti — APPROVED** [review]

File: tests/sdk/conftest.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+# Add the SDK src to the path to allow imports
+sdk_src_path = Path(__file__).parent.parent.parent / "sdks" / "python" / "src"
+sys.path.insert(0, str(sdk_src_path))
+
+# This is a bit of a hack to make the main conftest discoverable
+sys.path.insert(0, str(Path(__file__).parent.parent))
+
+from sdks.python.src.honcho.async_client.client import AsyncHoncho  # noqa: E402
+from sdks.python.src.honcho.client import Honcho  # noqa: E402
```
[the remark this answers is not in the document]
Comment (dr-frmr) [reply tests/sdk/conftest.py:17]: no those are necessary
